File: app/src/main/java/com/electricdreams/shellshock/ui/adapter/PaymentsHistoryAdapter.java
package com.electricdreams.shellshock.ui.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.electricdreams.shellshock.R;
import com.electricdreams.shellshock.core.data.model.PaymentHistoryEntry;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

public class PaymentsHistoryAdapter extends RecyclerView.Adapter<PaymentsHistoryAdapter.ViewHolder> {
    private final List<PaymentHistoryEntry> entries = new ArrayList<>();
    private final SimpleDateFormat dateFormat = new SimpleDateFormat("MMM d", Locale.getDefault());
    private OnItemClickListener onItemClickListener;

    public interface OnItemClickListener {
        void onItemClick(PaymentHistoryEntry entry, int position);
    }

    public void setOnItemClickListener(OnItemClickListener listener) {
        this.onItemClickListener = listener;
    }

    public void setEntries(List<PaymentHistoryEntry> newEntries) {
        entries.clear();
        entries.addAll(newEntries);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_payment_history, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        PaymentHistoryEntry entry = entries.get(position);
        
        // Format amount
        long amount = entry.getAmount();
        String sign = amount >= 0 ? "+" : "";
        holder.amountText.setText(String.format(Locale.getDefault(), "%s ₿%d", sign, Math.abs(amount)));
        
        // Set date
        holder.dateText.setText(dateFormat.format(entry.getDate()));
        
        // Set title based on amount (simple logic for now)
        if (amount > 0) {
            holder.titleText.setText("Cash In");
        } else {
            holder.titleText.setText("Cash Out");
        }

        holder.itemView.setOnClickListener(v -> {
            if (onItemClickListener != null) {
                onItemClickListener.onItemClick(entry, position);
            }
        });
    }

    @Override
    public int getItemCount() {
        return entries.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        final TextView amountText;
        final TextView dateText;
        final TextView titleText;

        ViewHolder(View view) {
            super(view);
            amountText = view.findViewById(R.id.amount_text);
            dateText = view.findViewById(R.id.date_text);
            titleText = view.findViewById(R.id.title_text);
        }
    }
}

File: app/src/main/java/com/electricdreams/shellshock/ui/adapter/PaymentsHistoryAdapter.kt
package com.electricdreams.shellshock.ui.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.electricdreams.shellshock.R
import com.electricdreams.shellshock.core.data.model.PaymentHistoryEntry
import com.electricdreams.shellshock.core.model.Amount
import java.text.SimpleDateFormat
import java.util.Locale

class PaymentsHistoryAdapter : RecyclerView.Adapter<PaymentsHistoryAdapter.ViewHolder>() {

    fun interface OnItemClickListener {
        fun onItemClick(entry: PaymentHistoryEntry, position: Int)
    }

    private val entries: MutableList<PaymentHistoryEntry> = mutableListOf()
    private val dateFormat = SimpleDateFormat("MMM d", Locale.getDefault())
    private var onItemClickListener: OnItemClickListener? = null

    fun setOnItemClickListener(listener: OnItemClickListener) {
        onItemClickListener = listener
    }

    fun setEntries(newEntries: List<PaymentHistoryEntry>) {
        entries.clear()
        entries.addAll(newEntries)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_payment_history, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val entry = entries[position]
        val context = holder.itemView.context

        // Display amount in the unit it was entered
        val formattedAmount = if (entry.getEntryUnit() != "sat") {
            val entryCurrency = Amount.Currency.fromCode(entry.getEntryUnit())
            val entryAmount = Amount(entry.enteredAmount, entryCurrency)
            entryAmount.toString()
        } else {
            val satAmount = Amount(entry.amount, Amount.Currency.BTC)
            satAmount.toString()
        }

        // Set amount with appropriate prefix
        val isPending = entry.isPending()
        val displayAmount = if (isPending) {
            formattedAmount // No + prefix for pending
        } else if (entry.amount >= 0) {
            "+$formattedAmount"
        } else {
            formattedAmount
        }
        holder.amountText.text = displayAmount

        // Set date
        holder.dateText.text = dateFormat.format(entry.date)

        // Set title based on status and payment type
        holder.titleText.text = when {
            isPending -> "Pending Payment"
            entry.isLightning() -> "Lightning Payment"
            entry.isCashu() -> "Cashu Payment"
            entry.amount > 0 -> "Cash In"
            else -> "Cash Out"
        }

        // Set icon based on payment type and status
        val iconRes = when {
            isPending -> R.drawable.ic_pending
            entry.isLightning() -> R.drawable.ic_lightning_bolt
            else -> R.drawable.ic_bitcoin
        }
        holder.icon.setImageResource(iconRes)

        // Set icon tint based on status
        val iconTint = if (isPending) {
            context.getColor(R.color.color_warning)
        } else {
            context.getColor(R.color.color_text_primary)
        }
        holder.icon.setColorFilter(iconTint)

        // Show status for pending payments
        if (isPending) {
            holder.statusText.visibility = View.VISIBLE
            holder.statusText.text = "Tap to resume"
            holder.statusText.setTextColor(context.getColor(R.color.color_warning))
        } else {
            holder.statusText.visibility = View.GONE
        }

        // Hide subtitle (payment type already shown in title)
        holder.subtitleText.visibility = View.GONE

        holder.itemView.setOnClickListener {
            onItemClickListener?.onItemClick(entry, position)
        }
    }

    override fun getItemCount(): Int = entries.size

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val amountText: TextView = view.findViewById(R.id.amount_text)
        val dateText: TextView = view.findViewById(R.id.date_text)
        val titleText: TextView = view.findViewById(R.id.title_text)
        val subtitleText: TextView = view.findViewById(R.id.subtitle_text)
        val statusText: TextView = view.findViewById(R.id.status_text)
        val icon: ImageView = view.findViewById(R.id.icon)
    }
}
